lib/Jsrt/Jsrt.cpp

```diff
@@ -1480,7 +1483,16 @@ CHAKRA_API JsHasProperty(_In_ JsValueRef object, _In_ JsPropertyIdRef propertyId
         *hasProperty = Js::JavascriptOperators::OP_HasProperty(object, ((Js::PropertyRecord *)propertyId)->GetPropertyId(), scriptContext) != 0;
 
         return JsNoError;
-    });
+    };
+
+    if (Js::JavascriptProxy::Is(object))
+    {
+        return ContextAPIWrapper<JSRT_MAYBE_TRUE>(internalHasProperty);
+    }
+    else
+    {
+        return ContextAPINoScriptWrapper(internalHasProperty);
```

Comment:
Tried looking around but I don't see if this can call into user script. So should be good to go. May be in other PR we can convert some hot JSRT APIs to similar mechanism.